crates/core/src/table/partition.rs

```diff
@@ -506,4 +525,75 @@ mod tests {
 
         assert!(pruner.parse_segments("invalid/path").is_err());
     }
+
+    #[test]
+    fn test_strip_single_quote_for_date_filter() {
+        for data_type in [DataType::Date32, DataType::Date64] {
+            let schema = Schema::new(vec![Field::new("date", data_type.clone(), false)]);
+
+            let cast_options = CastOptions {
+                safe: false,
+                format_options: Default::default(),
+            };
+
+            let value = StringArray::from(vec!["2023-01-01"]);
+            let value = cast_with_options(&value, &data_type, &cast_options).unwrap();
+
+            let filter_str = "date = '2023-01-01'";
+            let filter = PartitionFilter::try_from((filter_str, &schema));
+            assert!(filter.is_ok());
+            let filter = filter.unwrap();
+            assert_eq!(filter.field.name(), "date");
+            assert_eq!(filter.operator, Operator::Eq);
+            assert_eq!(filter.value.get().0.len(), 1);
+            assert_eq!(filter.value.get().0, value.get().0);
+        }
+    }
+
+    #[test]
+    fn test_strip_single_quote_for_string_filter() {
+        for data_type in [DataType::Utf8, DataType::Utf8View, DataType::LargeUtf8] {
+            let schema = Schema::new(vec![Field::new("category", data_type.clone(), false)]);
+
+            let cast_options = CastOptions {
+                safe: false,
+                format_options: Default::default(),
+            };
+
+            let value = StringArray::from(vec!["foo"]);
+            let value = cast_with_options(&value, &data_type, &cast_options).unwrap();
+
+            let filter_str = "category!='foo'";
+            let filter = PartitionFilter::try_from((filter_str, &schema));
+            assert!(filter.is_ok());
+            let filter = filter.unwrap();
+            assert_eq!(filter.field.name(), "category");
+            assert_eq!(filter.operator, Operator::Ne);
+            assert_eq!(filter.value.get().0.len(), 1);
+            assert_eq!(filter.value.get().0, value.get().0)
+        }
+    }
+
+    #[test]
+    fn test_do_not_strip_one_side_single_quoted_value_for_filter() {
+        let data_type = DataType::Utf8;
+        let schema = Schema::new(vec![Field::new("category", data_type.clone(), false)]);
+
+        let cast_options = CastOptions {
+            safe: false,
+            format_options: Default::default(),
+        };
+
+        let value = StringArray::from(vec!["'foo"]);
```

Comment:
what i meant in the previous comment was: shouldn't this fail due to invalid sql syntax? can you check how these work with other sql engines? for e.g., how would datafusion treat this when query parquet table?

Reply:
im not sure I follow
how the query would look like?

For sure this will work properly:
`SELECT * FROM table where part='2022-01-01'`
but
`SELECT * FROM table where part='2022-01-01`
will fail when parsing the query.

Reply:
as for strings this should result in an error?
`SELECT * FROM table where part=A`
and this should pass:
`SELECT * FROM table where part=A`?

Reply:
i'm not sure about this, shouldn't this be the responsibility of a query engine to parse this?
The partition pruner could be implemented specifically for a query engine first?

Reply:
@xushiyan what if we flip it a bit and make 
PartitionFilter::try_from() take a 3 element tuple containing key, operator, value instead?
Then we can (based on the schema) simply assume the type and don't need worry about the single quotes.

so:
`PartitionFilter::try_from(("part", "=", "A"))`
`PartitionFilter::try_from(("part", "=", "2022-01-01"))`
`PartitionFilter::try_from(("part", "IN", ["2022-01-01", "2023-02-02"]))`

Then we simplify and leave the parsing to the user or system that uses that, but it's simple enough not to be frustrating to use.

Reply:
@xushiyan please take a look and let me know what you think.
Imo this simplifies a lot on hudi-rs side, still pretty ok from the user's point of view.

Now the other way around it should be easy to generate these filter strings if required by some engine etc.

Reply:
yup the approach looks good. it's good to start with something simple, then iterate on it. getting back to this. will review later